protocols/bgp/server/fsm_address_family.go

```diff
@@ -138,7 +140,7 @@ func (f *fsmAddressFamily) getSessionAttrs() routingtable.SessionAttrs {
 }
 
 func (f *fsmAddressFamily) bmpInit() {
-	f.adjRIBIn = f.fsm.peer.adjRIBInFactory.New(filter.NewAcceptAllFilterChain(), &routingtable.ContributingASNs{}, f.getSessionAttrs())
+	f.adjRIBIn = f.fsm.peer.adjRIBInFactory.New(filter.NewAcceptAllFilterChain(), f.fsm.peer.vrf, f.getSessionAttrs())
```

Comment:
Should we just pass a freshly create VRF here to make 100% sure to adhere to what the router sent?

Reply:
I don't think so. The VRF should point to a VRF that was created for the specific RD that was in a peer up notification.